oversweet-api/src/main/java/com/depromeet/oversweet/franchise/controller/FranchiseController.java

```diff
@@ -33,4 +39,22 @@ public ResponseEntity<DataResponse<List<FranchiseInfo>>> getFranchiseByKeyword(@
         return ResponseEntity.ok()
                 .body(DataResponse.of(HttpStatus.OK, "검색 키워드로 해당하는 프랜차이즈 목록 조회 성공", response));
     }
+
+    @Operation(summary = "레디스에 프랜차이즈 목록 세팅", description = "레디스에 프랜차이즈 목록 세팅 API")
+    @ApiResponses(@ApiResponse(responseCode = "200", description = "레디스에 프랜차이즈 목록 세팅 성공"))
+    @PostMapping("redis")
+    public ResponseEntity<MessageResponse> createFranchiseAtRedis() {
+        franchiseRedisService.createFranchiseAtRedis();
+        return ResponseEntity.ok()
+                .body(MessageResponse.of(OK, "레디스에 프랜차이즈 목록 세팅 성공"));
+    }
+
+    @Operation(summary = "레디스에 저장된 프랜차이즈 목록 조회", description = "레디스에 저장된 프랜차이즈 목록 조회 API")
+    @ApiResponses(@ApiResponse(responseCode = "200", description = "레디스에 저장된 프랜차이즈 목록 조회 성공"))
+    @GetMapping("redis")
+    public ResponseEntity<DataResponse<List<FranchiseInfo>>> getFranchiseAtRedis() {
+        final List<FranchiseInfo> franchise = franchiseRedisService.getFranchiseAtRedis();
+        return ResponseEntity.ok()
+                .body(DataResponse.of(OK, "레디스에 저장된 프랜차이즈 목록 조회 성공", franchise));
+    }
```

Comment:
개인적으로 목록을 세팅하는 API와 조회하는 API을 따로 호출하도록 할 이유가 있을까?? 라는 생각이 들어요

목록 조회하는 API 하나만 호출해도,
Service 계층에서 DB에 모든 FranchiseEntity List 을 Redis에 저장하고

DB에서 조회한 List FranchiseEntity을 그대로 반환 하도록 하면 2개의 API을 하나의 API로 축소할 수 있지 않을까요??

```suggestion
@Service
@RequiredArgsConstructor
public class FranchiseRedisService {
    private static final String FRANCHISE = "Franchise";
    private final RedisRepository redisRepository;
    private final FindAllFranchiseRepository findAllFranchiseRepository;

    public List<FranchiseInfo> getFranchises() {
        List<FranchiseInfo> franchiseInfos = redisRepository.getData(FRANCHISE, FranchiseInfo.class);

        if (franchiseInfos == null || franchiseInfos.isEmpty()) {
            final List<FranchiseEntity> findFranchises = findAllFranchiseRepository.findAll();
            franchiseInfos = findFranchises.stream()
                    .map(FranchiseInfo::of)
                    .toList();
            redisRepository.saveData(FRANCHISE, franchiseInfos);
        }

        return franchiseInfos;
    }
}

```


이렇게 가져가면 Redis에 데이터가 있는 경우 Cache Hit 가 발생하고, 없으면 직접 DB에서 조회하는 방향으로 하면 어떨까요?